hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/storage/AsyncInventoryStorage.java

```diff
@@ -0,0 +1,444 @@
+                    .toString();
+            rPojo = new org.hawkular.inventory.api.model.Resource.Blueprint(getInventoryId(resource), resourceTypePath,
+                    resource.getProperties());
+
+            StringBuilder parentPath = new StringBuilder();
+            Resource<?, ?, ?, ?, ?> parent = resource.getParent();
+            while (parent != null) {
+                String resourceIdPath = "/" + Util.urlEncode(parent.getID().getIDString());
+                parentPath.insert(0, resourceIdPath);
+                parent = parent.getParent();
+            }
+
+            CanonicalPath parentCanonicalPath = parentPath.length() == 0 ? newPathPrefix().get()
+                    : CanonicalPath.fromPartiallyUntypedString(parentPath.toString(), newPathPrefix().get(),
+                            org.hawkular.inventory.api.model.Resource.class);
+
+            relationshipOrEntity(parentCanonicalPath.toString(), rPojo,
+                    org.hawkular.inventory.api.model.Resource.class);
+
+            String resourcePath = parentPath.toString() + "/" + Util.urlEncode(resource.getID().getIDString());
+            CanonicalPath resourceCanonicalPath = CanonicalPath.fromPartiallyUntypedString(resourcePath,
+                    newPathPrefix().get(), org.hawkular.inventory.api.model.Resource.class);
+
+            // now that the resource is registered, immediately register its configuration
+            Collection<? extends ResourceConfigurationPropertyInstance<?>> resConfigInstances = resource
+                    .getResourceConfigurationProperties();
+            if (resConfigInstances != null && !resConfigInstances.isEmpty()) {
+                // get the payload in JSON format
+                StructuredData.MapBuilder structDataBuilder = StructuredData.get().map();
+                for (ResourceConfigurationPropertyInstance<?> resConfigInstance : resConfigInstances) {
+                    structDataBuilder.putString(resConfigInstance.getID().getIDString(), resConfigInstance.getValue());
+                }
+
+                org.hawkular.inventory.api.model.DataEntity.Blueprint<Resources.DataRole> dataEntity = //
+                new org.hawkular.inventory.api.model.DataEntity.Blueprint<>(Resources.DataRole.configuration,
+                        structDataBuilder.build(), null);
+
+                relationshipOrEntity(resourceCanonicalPath.toString(), dataEntity, DataEntity.class);
+            }
+
+            Collection<? extends MetricInstance<?, ?, ?>> metricInstances = resource.getMetrics();
+            for (MetricInstance<?, ?, ?> metric : metricInstances) {
+                metric(metric);
+                CanonicalPath metricPath = newPathPrefix().metric(getInventoryId(metric)).get();
+                Relationship.Blueprint bp = new Relationship.Blueprint(Direction.outgoing, incorporates.toString(),
+                        metricPath, metric.getProperties());
```

Comment:
`metric.getProperties()` is probably not what you want on the relationship. What you're doing there is that you define the properties of the relationship itself, so I'm not sure if the relationship should hold a copy of properties stored at its target entity.